scio-bigquery/src/it/scala/com/spotify/scio/bigquery/BigQueryIT.scala

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2016 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.bigquery
+
+import com.spotify.scio.bigquery.types._
+import org.scalatest.Inspectors.forAll
+import org.scalatest._
+
+object BigQueryIT {
+  val tableRef = "bigquery-public-data:samples.shakespeare"
+  val legacyQuery =
+    "SELECT word, word_count FROM [bigquery-public-data:samples.shakespeare] LIMIT 10"
+  val sqlQuery =
+    "SELECT word, word_count FROM `bigquery-public-data.samples.shakespeare` LIMIT 10"
+
+  @BigQueryType.fromTable("bigquery-public-data:samples.shakespeare")
+  class Shakespeare
+
+  @BigQueryType.fromQuery(
+    "SELECT word, word_count FROM `bigquery-public-data.samples.shakespeare` LIMIT 10")
+  class WordCount
+
+}
+
+/**
+ * BigQuery integration test with mock data and equality check for query result.
+ */
+class BigQueryMockIT extends FlatSpec with Matchers with BigQueryITUtils {
+
+  import BigQueryIT._
+
+  private def shakespeare(w: String, wc: Long, c: String, cd: Long) =
+    TableRow("word" -> w, "word_count" -> wc, "corpus" -> c, "corpus_date" -> cd)
+
```

Comment:
Would it be worth adding some helper so that users don't have to define their own functions / create table rows manually? Realize it's not much overhead really, but even just so that people don't have to work with TableRow directly. Maybe something like this? Would have to define this for up to 23 type params but could easily be generated similar to MultiJoin, and it's slightly "safer" in that users won't be able to pass in TableRows with different arity.

```
mockTable(tableRef).withData(
  Data("word", "word_count", "corpus", "corpus_date")
    .add("i", 10, "kinglear", 1600)
    .add("thou", 20, "kinglear", 1600)
  )
)
```